lib/_http_client.js

```diff
@@ -636,7 +636,11 @@ ClientRequest.prototype.onSocket = function onSocket(socket) {
 function onSocketNT(req, socket) {
   if (req.aborted) {
     // If we were aborted while waiting for a socket, skip the whole thing.
-    socket.emit('free');
+    if (req.socketPath || !req.agent) {
+      socket.destroy();
+    } else {
+      socket.emit('free');
+    }
```

Comment:
I think we should always `destroy()`, I do not see a way for this socket to be reused even if there is an agent.